File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.java

class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int ans=0;
        int sum=0;
        for (int i=0;i<k;i++){
            sum += cardPoints[i];
        }
        ans = sum;
        int i=k-1, j= cardPoints.length-1;
        while (i>=0){
            sum -= (cardPoints[i]-cardPoints[j]);
            ans =Math.max(ans, sum);
            i--;
            j--;
        }
        return ans;
    }
}

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        
        int n = cardPoints.size();
        int x = n-k;
        
        int sum=0;
        
        for(int i=0;i<n;i++)
            sum+=cardPoints[i];
        
        int mx=0;
        
        int kwindow=0;
        
        for(int i=0;i<x;i++)
        {
            kwindow+=cardPoints[i];
        }
        
        mx=max(mx,sum-kwindow);
        
        for(int i=x;i<n;i++)
        {
            kwindow-=cardPoints[i-x];
            kwindow+=cardPoints[i];
            mx=max(mx,sum-kwindow);
        }
        
        return mx;
    }
};